src/Goerli-DssSpell.sol

```diff
@@ -23,27 +23,119 @@ import "dss-exec-lib/DssAction.sol";
 
 import { DssSpellCollateralOnboardingAction } from "./Goerli-DssSpellCollateralOnboarding.sol";
 
+interface RegistryLike {
+    function list() external view returns (bytes32[] memory);
+    function pip(bytes32 ilk) external view returns (address);
+    function class(bytes32 ilk) external view returns (uint256);
+}
 
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     // Provides a descriptive tag for bot consumption
     string public constant override description = "Goerli Spell";
 
-    // Turn office hours off
+    // Math
+    uint256 constant internal MILLION  = 10 ** 6;
+    uint256 constant internal BILLION  = 10 ** 9;
+
+    // Many of the settings that change weekly rely on the rate accumulator
+    // described at https://docs.makerdao.com/smart-contract-modules/rates-module
+    // To check this yourself, use the following rate calculation (example 8%):
+    //
+    // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
+    //
+    // A table of rates can be found at
+    //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
+    //
+
+    // --- Rates ---
+    uint256 constant ZERO_PCT_RATE           = 1000000000000000000000000000;
+    uint256 constant ZERO_ZERO_FIVE_PCT_RATE = 1000000000015850933588756013;
+    uint256 constant TWO_TWO_FIVE_PCT_RATE   = 1000000000705562181084137268;
+    uint256 constant THREE_TWO_FIVE_PCT_RATE = 1000000001014175731521720677;
+    uint256 constant FOUR_FIVE_PCT_RATE      = 1000000001395766281313196627;
+
+    address constant public ORACLE_WALLET01 = 0x4D6fbF888c374D7964D56144dE0C0cFBd49750D3;
```

Comment:
I wonder if either of these should be added to `./src/test/addresses_deployers.sol`

Reply:
I took this approach first, but the deployers check ensures the deployers are no longer `wards`.  In goerli, this check is basically short circuited with a `continue`.

Reply:
That makes sense. We should probably add these addresses to `spell-mainnet`'s deployer list to ensure they aren't wards there.